packages/flutter_tools/lib/src/commands/build_ios.dart

```diff
@@ -293,20 +291,24 @@ abstract class _BuildIOSSubCommand extends BuildSubCommand {
       final File precompilerTrace = globals.fs.directory(buildInfo.codeSizeDirectory)
         .childFile('trace.$arch.json');
 
-      final Directory outputAppDirectoryCandidate = _outputAppDirectory(result.output);
+      final String? resultOutput = result.output;
+      if (resultOutput == null) {
+        throwToolExit('Could not find app to analyze code size');
+      }
+      final Directory outputAppDirectoryCandidate = _outputAppDirectory(resultOutput);
 
-      Directory appDirectory;
+      Directory? appDirectory;
       if (outputAppDirectoryCandidate.existsSync()) {
         appDirectory = outputAppDirectoryCandidate.listSync()
             .whereType<Directory>()
-            .firstWhere((Directory directory) {
+            .where((Directory directory) {
           return globals.fs.path.extension(directory.path) == '.app';
-        }, orElse: () => null);
+        }).first;
```

Comment:
To preserve the original intent, would it make sense to use `firstWhereOrNull` from `package:collection`?

Reply:
`flutter_tool` doesn't have a direct dependency on `package:collection`.  Fortunately this list should be small.